extras/module-node/README.rst

```diff
@@ -0,0 +1,76 @@
+=====================================
+Node.js-like module loading framework
+=====================================
+
+This directory contains an example module resolution and loading framework and
+``require`` implementation based on the Node.js module system:
+
+* https://nodejs.org/api/modules.html
+
+The application needs only to provide the module resolution and loading logic:
+
+* Add ``duk_module_node.c`` to list of C sources to compile.
+
+* Ensure ``duk_module_node.h`` is in the include path.
+
+* Include the extra header in calling code and initialize the bindings::
+
+      #include "duktape.h"
+      #include "duk_module_node.h"
+
+      /* After initializing the Duktape heap or when creating a new
+       * thread with a new global environment:
+       */
+      duk_push_object(ctx);
+      duk_push_c_function(ctx, cb_resolve_module, DUK_VARARGS);
+      duk_put_prop_string(ctx, -2, "resolve");
+      duk_push_c_function(ctx, cb_load_module, DUK_VARARGS);
+      duk_put_prop_string(ctx, -2, "load");
+      duk_module_node_init(ctx);
+
+  Do not call ``duk_module_node_init()`` more than once for the same global
+  environment.  Doing so will put the module system in an inconsistent state.
+
+* The resolve callback is a Duktape/C function which takes the string passed
+  to `require()` and resolves it to a canonical module ID (for Node.js this is
```

Comment:
Yup, same here: it's somewhat annoying to have both Markdown and RST documents in the repo. I started with RST so I've only used Markdown for a few documents people usually expect to be Markdown (like README.md). (Though, Github does support RST for README, CONTRIBUTING, etc too.)